### ISSUE TITLE
ERROR: The Compose file './docker-compose.yml' is invalid because: services.web.volumes contains an invalid type, it should be a string

### ISSUE THREAD

**Issue by vishnutm** [docker/compose#7421]
Trying to setup docker-compose file and it doesn't work 

Followed code for docker-compose file

![Screenshot from 2020-05-03 22-31-31](https://user-images.githubusercontent.com/30982642/80920512-f8a9bc80-8d8d-11ea-88ec-9482b7ac1fc0.png)

Docker  Version:           19.03.8
docker-compose version 1.24.0, build 0aa59064

**Comment by thaJeztah** [comment]
looks like there's a space in the line

```
- .: /app
```

**Comment by vishnutm** [comment]
Thanks. it worked

**Comment by ikidnapmyself** [comment]
in my case, i updated docker to following on MacOS Catalina:

```
$ docker-compose version
docker-compose version 1.27.4, build 40524192
docker-py version: 4.3.1
CPython version: 3.7.7
OpenSSL version: OpenSSL 1.1.1g  21 Apr 2020
```

```
$ docker version
Client: Docker Engine - Community
 Cloud integration  0.1.18
 Version:           19.03.13
 API version:       1.40
 Go version:        go1.13.15
 Git commit:        4484c46d9d
 Built:             Wed Sep 16 16:58:31 2020
 OS/Arch:           darwin/amd64
 Experimental:      false

Server: Docker Engine - Community
 Engine:
  Version:          19.03.13
  API version:      1.40 (minimum version 1.12)
  Go version:       go1.13.15
  Git commit:       4484c46d9d
  Built:            Wed Sep 16 17:07:04 2020
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.3.7
  GitCommit:        8fba4e9a7d01810a393d5d25a3621dc101981175
 runc:
  Version:          1.0.0-rc10
  GitCommit:        dc9208a3303feef5b3839f4323d9beb36df0a9dd
 docker-init:
  Version:          0.18.0
  GitCommit:        fec3683
```
<img width="1128" alt="resim" src="https://user-images.githubusercontent.com/13779866/95297937-f0e1cd00-0883-11eb-94bd-716977637c90.png">

until I update Docker, all the workspaces were working as expected but then I faced this error. In order to solve it, I fixed my docker-compose.yml file by setting .env file. I added missing constants of error triggering lines.  actually I am using only nginx, mysql, redis and php-worker but however it seems some undefined contstants causes this error.

Example:
<img width="601" alt="resim" src="https://user-images.githubusercontent.com/13779866/95297139-a27ffe80-0882-11eb-979d-05c930c901d8.png">

Basically, regenerated my .env file and it completed my missing constant defaults.
